station/clients/airflow/docker_trains.py

```diff
@@ -3,47 +3,127 @@
 from typing import Any
 import os
 from datetime import datetime
+import json
 
 from .client import airflow_client
 from station.app.crud.crud_docker_trains import docker_trains
 from station.app.crud.crud_train_configs import docker_train_config
-from station.app.schemas.docker_trains import DockerTrainExecution
+from station.app.schemas.docker_trains import DockerTrainExecution, DockerTrainSavedExecution, DockerTrainAirflowConfig, DockerTrainState
+from station.app.models.docker_trains import DockerTrainState as dts_model, DockerTrainExecution as dte_model
 
 
-# TODO rework this !!!
-def run_train(db: Session, train_id: Any, run_config: DockerTrainExecution):
+def update_state(db, db_train, run_time) -> DockerTrainState:
     """
-    Execute a PHT 1.0 docker train using a configured airflow instance
+    Update the train state object corresponding to the train
+    :param db: database session
+    :param db_train: train object
+    :param run_time: time when run is triggered
+    :return: train state object
+    """
+    train_state = db.query(dts_model).filter(dts_model.train_id == db_train.id).first()
+    if train_state:
+        train_state.last_execution = run_time
+        train_state.num_executions += 1
+        train_state.status = 'active'
+    else:
+        print("No train state assigned.")
+    db.add(train_state)
+    db.commit()
+    db.refresh(train_state)
 
-    :param db:
-    :param train_id: identifier of the train
-    :param run_config: given config_id or config_json can be used for running train
+    return train_state
+
+
+def validate_run_config(db: Session, train_id: Any, execution_params: DockerTrainExecution) -> DockerTrainAirflowConfig:
+    """
+    Validate the config used for the triggered run
+    :param db: database session
+    :param train_id: train id of the train to run
+    :param execution_params: includes the config_id of the config to use or the specified config
     :return:
     """
-
-    if run_config.config_id != "default":
-        config = docker_train_config.get(db, run_config.config_id)
-    elif run_config.config_json:
-        config = run_config.config_json
+    # Extract config by id if given
+    if execution_params.config_id != "default":
+        config_general = docker_train_config.get(db, execution_params.config_id)
+        try:
+            config = config_general.airflow_config
+        except:
+            raise HTTPException(status_code=400, detail="No airflow config given by this id.")
+    # Extract config as defined in the execution
+    elif execution_params.config_json:
+        config = json.loads(execution_params.config_json.json())
```

Comment:
just use dict instead of config -> json -> json load -> dict
`execution_params.config_json.dict()`